azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/options/AzureTemplateOptions.java

```diff
@@ -100,44 +145,52 @@ public void copyTo(TemplateOptions to) {
          eTo.availabilitySet(availabilitySet);
          eTo.availabilitySet(availabilitySetName);
+         eTo.dataDisks(dataDisks);
       }
    }
 
    @Override
-   public int hashCode() {
-      return Objects.hashCode(super.hashCode(), virtualNetworkName, subnetId, blob, availabilitySet,
-            availabilitySetName);
-   }
+   public boolean equals(Object o) {
+      if (this == o) return true;
+      if (!(o instanceof AzureTemplateOptions)) return false;
+      if (!super.equals(o)) return false;
 
-   @Override
-   public boolean equals(Object obj) {
-      if (this == obj) {
-         return true;
-      }
-      if (!super.equals(obj)) {
+      AzureTemplateOptions that = (AzureTemplateOptions) o;
+
+      if (virtualNetworkName != null ? !virtualNetworkName.equals(that.virtualNetworkName) : that.virtualNetworkName != null)
          return false;
-      }
-      if (getClass() != obj.getClass()) {
+      if (subnetId != null ? !subnetId.equals(that.subnetId) : that.subnetId != null) return false;
+      if (blob != null ? !blob.equals(that.blob) : that.blob != null) return false;
+      if (availabilitySet != null ? !availabilitySet.equals(that.availabilitySet) : that.availabilitySet != null)
          return false;
-      }
-      AzureTemplateOptions other = (AzureTemplateOptions) obj;
-      return super.equals(other)
-            && equal(this.virtualNetworkName, other.virtualNetworkName)
-            && equal(this.subnetId, other.subnetId)
-            && equal(this.blob, other.blob)
-            && equal(this.availabilitySet, other.availabilitySet)
-            && equal(this.availabilitySetName, other.availabilitySetName);
+      if (availabilitySetName != null ? !availabilitySetName.equals(that.availabilitySetName) : that.availabilitySetName != null)
+         return false;
+      return dataDisks != null ? dataDisks.equals(that.dataDisks) : that.dataDisks == null;
    }
 
    @Override
-   public Objects.ToStringHelper string() {
-      Objects.ToStringHelper toString = super.string().omitNullValues();
-      toString.add("virtualNetworkName", virtualNetworkName);
-      toString.add("subnetId", subnetId);
-      toString.add("blob", blob);
-      toString.add("availabilitySet", availabilitySet);
-      toString.add("availabilitySetName", availabilitySetName);
-      return toString;
+   public int hashCode() {
+      int result = super.hashCode();
+      result = 31 * result + (virtualNetworkName != null ? virtualNetworkName.hashCode() : 0);
+      result = 31 * result + (subnetId != null ? subnetId.hashCode() : 0);
+      result = 31 * result + (blob != null ? blob.hashCode() : 0);
+      result = 31 * result + (availabilitySet != null ? availabilitySet.hashCode() : 0);
+      result = 31 * result + (availabilitySetName != null ? availabilitySetName.hashCode() : 0);
+      result = 31 * result + (dataDisks != null ? dataDisks.hashCode() : 0);
+      return result;
+   }
+
+   @Override
+   public String toString() {
```

Comment:
I've implemented dozen of `*TemplateOptions` and I've never noticed `string()` :) 
Thanks!